rust/hyperlane-base/src/settings/chains.rs

```diff
@@ -140,6 +143,37 @@ impl ChainConf {
         .context(ctx)
     }
 
+    /// Try to convert the chain setting into a Merkle Tree Hook contract
+    pub async fn build_merkle_tree_hook(
+        &self,
+        metrics: &CoreMetrics,
+    ) -> Result<Box<dyn MerkleTreeHook>> {
+        let ctx = "Building merkle tree hook";
+        // TODO: if the merkle tree hook is set for sealevel, it's still a mailbox program
+        // that the connection is made to using the pda seeds, which will not be usable.
+        let address = self
```

Comment:
I don't like the silently different behavior tbh

I think we should make merkle_tree_hook mandatory for now, and just change Sealevel in the JSON configs to specify the Mailbox address as the merkleTreeHook

Eventually I could see us wanting to make hooks in general optional, but that feels out of the scope of what this PR should do

Reply:
just flagging that even though we have dummy implementations for the merkle tree indexers on sealevel, we won't be covering sealevel in e2e so panics could still occur